daemon/daemon.go

```diff
@@ -916,6 +917,59 @@ func createPrefixLengthCounter() *counter.PrefixLengthCounter {
 	return counter
 }
 
+func (d *Daemon) prepareAllocationCIDR(family datapath.NodeAddressingFamily) (routerIP net.IP, err error) {
+	// Reserve the IPv4 external node IP within the allocation range if
+	// required.
+	allocRange := family.AllocationCIDR()
+	nodeIP := family.PrimaryExternal()
+	if allocRange.Contains(nodeIP) {
+		err = d.ipam.AllocateIP(nodeIP)
+		if err != nil {
+			err = fmt.Errorf("Unable to allocate external IPv4 node IP %s from allocation range %s: %s",
+				nodeIP, allocRange, err)
+			return
+		}
+	}
+
+	routerIP = family.Router()
+	if routerIP != nil && !allocRange.Contains(routerIP) {
+		log.Warningf("Detected allocation CIDR change to %s, previous router IP %s", allocRange, routerIP)
+
+		// The restored router IP is not part of the allocation
+		// range. This indicates that the allocation range has
+		// changed.
+
+		// Remove the old host device and
+		var link netlink.Link
+		link, err = netlink.LinkByName(defaults.HostDevice)
```

Comment:
replace with `option.Config.HostDevice`

Reply:
@aanm Are you sure that you want to delete the flannel device?

Reply:
Can we introduce the flag `doNotDelOldHostDevice` (set in the flannel mode) to the method, and then replace as suggested ^^?

Reply:
I've added the check. We can work on a `doNotDelOldHostDevice` if we have more than one use case. I'm not sure about IPVLAN yet.